Close #500 - Add missing visitINVOKEDYNAMIC override in TypeFrameModelingVisitor

wrprice opened the pull request:
Prevents underflowing the stack (reaching `<bottom>`) when encountering
INVOKEDYNAMIC.  Without the override, parent class logic was consuming
a stack frame but not pushing the return type from the INDY call.

Symptom was IllegalArgumentException ("Not a reference type: &lt;bottom&gt;")
from FindUnrelatedTypesInGenericContainer. See issue #500 



----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

wrprice commented:
@ThrawnCA you're technically correct, because the MCVE relies on under-the-hood `javac` changes in Java 9.  Did you mean that merely as an observation, or are you requesting an alternative/improvement?

As written, the test case verifies that there are no bugs found (or exceptions thrown) in the MCVE under _whatever version the build is using_, and that behavior should hold for _all_ Java >= 5.  Nothing in the MCVE _source_ requires Java 9 features.  Remove the new `@Override` in the production code and the test indeed fails on JDK 9.  (I wasn't able to build cleanly, prior to any changes, on JDK 10.  I assume that's expected?)

I suppose I can try to find an explicit lambda or something that would compile down to a similar INDY on Java 8 and hopefully trigger the same problem.  That would be a good additional test case, but I have lambdas and method references all over my production code and this was the only case (so far) where I've hit the bug.

By the way, I did consider putting the example class under `spotbugsTestCases/src/java9/`, but based on the Gradle project configuration those are only compiled in the case...

    if (System.getProperty("java.specification.version") == "9")

Lexical string ordering aside, maybe that was intended to be a `>=` relationship?  Or will these classes (and presumably their tests) stop working in the future when the project's CI no longer includes Java 9?

ThrawnCA commented:
Well, it would be nice to have a test that works on Java 8 too, but the change looks OK and it at least *has* a test (which many features lack), so I've approved it.

I'm not sure whether the `java9` test cases are supposed to be `>= java 9`, but probably.

wrprice commented:
Found a way to reproduce on Java 8, albeit with two lines of code that really warrant their own dodgy code detector.  Updated PR and squashed; please review again.

- Introduced second minimal example method into Issue500.java, uses a method reference.
- Above required moving Issue500.java from `ghIssues` package to `lambdas` because only certain packages are allowed to compile for target > 1.7
- Production fix is unchanged from prior rev.
